doc/whats-new.rst

```diff
@@ -26,6 +26,10 @@ Bug fixes
 
 - Fix to ensure xarray works with h5netcdf v0.3.0 for arrays with ``dtype=str``
   (:issue:`953`). By `Stephan Hoyer <https://github.com/shoyer>`_.
+- ``Dataset.__dir__()`` (i.e. the method python calls to get autocomplete options) failed
+  if one of the ``Dataset``'s keys was not a string (:issue:`852`). By
```

Comment:
Just a note, in restructured text this breaks:

    ``Dataset``'s

You need whitespace or punctuation after the backticks.